optuna/samplers/_tpe/sampler.py

```diff
@@ -816,6 +914,8 @@ def _get_multivariate_observation_pairs(
     scores = []
     values: Dict[str, List[Optional[float]]] = {param_name: [] for param_name in param_names}
     for trial in study.get_trials(deepcopy=False, states=(TrialState.COMPLETE, TrialState.PRUNED)):
+        if any([param_name not in trial.params for param_name in param_names]):
+            continue
```

Comment:
Might be helpful to leave a comment for future maintainers to why this condition is needed since it's not clear at first sight.